tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+#define MAX_DATA_SIZE 1000
+
+/* Implementation of safe malloc which returns NULL if the reuested size is 0.*/
+void *safeMalloc(size_t xWantedSize) {
+  if(xWantedSize == 0) {
+    return NULL;
+  }
+  uint8_t byte;
+  return byte ? malloc(xWantedSize) : NULL;
+}
+
+/* This modeling is required because otherwise CBMC takes
+too much time if assigned to minimumBufferSizes. */
+typedef struct _responseHandle
+{
+    struct _httpsResponse RespHandle;
+    char data[MAX_DATA_SIZE];
+} _resHandle_t;
+
+typedef struct _requestHandle
+{
+    struct _httpsRequest ReqHandle;
+    char data[MAX_DATA_SIZE];
+} _reqHandle_t;
+
+typedef struct _connectionHandle
+{
+    struct _httpsConnection pConnHandle;
+    char data[MAX_DATA_SIZE];
+} _connHandle_t;
+
+/* Models the third party HTTP Parser. */
+size_t http_parser_execute (http_parser *parser,
```

Comment:
I'm depending on Sarena to know that this is a reasonable model of the parser.

Reply:
This is fine for proof purposes.